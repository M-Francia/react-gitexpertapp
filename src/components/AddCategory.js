import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => { 
    
    const [inputValue, setinputValue] = useState(''); //

    const handleInputChange = (e) => {
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) => {
    
        e.preventDefault(); //Para que no reviente

        if( inputValue.trim().length > 2 ){
            setCategories( categories => [inputValue, ...categories] );
            setinputValue('');
        }

    }

    return(
        <form onSubmit={handleSubmit}>
        <h1>{inputValue}</h1>
        <p>{inputValue}</p>
        <input
            type="text"
            value={inputValue}
            onChange={ handleInputChange }
        />
        </form>
    )

 }

 AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}