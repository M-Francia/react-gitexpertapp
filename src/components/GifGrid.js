import React  from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
//import { getGifts } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {


    const {data:images , loading} = useFetchGifs(category);

    //const [images, setImages] = useState([]);

    return (
    <>
        <h3>{category}</h3>
        
        <div className='card-grid'>
            {loading && <p>Loading...</p>}
                {
                    images.map( (img) =>{
                        return <GifGridItem
                            key={img.id} 
                            { ...img }
                        />
                    })
                }
        </div>
    </>
  )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
