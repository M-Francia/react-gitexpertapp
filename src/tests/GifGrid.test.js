import { shallow } from "enzyme"
import { GifGrid } from "../components/GifGrid"
import { useFetchGifs } from "../hooks/useFetchGifs";

jest.mock("../hooks/useFetchGifs"); //Fingimos la llamada a esta funcion

describe('<GifGrid /> tests', () => { 
    
    const category = "test";
    
    test('should display the item', () => { 
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();

     });

     test('should have the info after loading useFetch()', () => { 
         
        const gifs =[{
            id:'ABC',
            url: 'https://algo.com/img.jpg',
            title:'cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists() ).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

      });



 })