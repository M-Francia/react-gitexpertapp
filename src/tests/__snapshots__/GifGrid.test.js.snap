// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> tests should display the item 1`] = `
<Fragment>
  <h3>
    test
  </h3>
  <div
    className="card-grid"
  >
    <p>
      Loading...
    </p>
  </div>
</Fragment>
`;

exports[`<GifGrid /> tests should have the info after loading useFetch() 1`] = `
<Fragment>
  <h3>
    test
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://algo.com/img.jpg"
    />
  </div>
</Fragment>
`;
